src/main/java/org/apache/bcel/generic/Type.java

```diff
@@ -365,6 +366,24 @@ public int hashCode() {
         return type ^ signature.hashCode();
     }
 
+    static String internalTypeNameToSignature(final String internalTypeName) {
+        if (StringUtils.isEmpty(internalTypeName) || StringUtils.equalsAny(internalTypeName, "B", "C", "D", "F", "I", "J", "S", "Z")) {
+            return internalTypeName;
```

Comment:
I'm not crazy about this magic list of strings. Could we reuse or refactor `org.apache.bcel.generic.InstructionFactory.SHORT_NAMES` (by making it public)?

Reply:
The array has different values

Reply:
I couldn't find any array containing these exact short names. The mentioned [SHORT_NAMES](https://github.com/apache/commons-bcel/blob/d256a461b40e91569961ae02ad3f33dd51814024/src/main/java/org/apache/bcel/generic/InstructionFactory.java#L55) misses `Z`, and `J`, but contains `L` additionally. The `SHORT_TYPE_NAMES` in [Consts](https://github.com/apache/commons-bcel/blob/master/src/main/java/org/apache/bcel/Const.java#L2851) and [Constants](https://github.com/apache/commons-bcel/blob/master/src/main/java/org/apache/bcel/Constants.java#L2364) both contain `V` and `ILLEGAL_TYPE` unnecessarily. There are constants for the short names individually in [ElementValue](https://github.com/apache/commons-bcel/blob/master/src/main/java/org/apache/bcel/classfile/ElementValue.java#L65), but I'm not sure using those would help a lot.

Reply:
Hi all,
I'm looking for clarity when reading the code. A hard coded array of strings is bad. Having a good name for a constant would help, and SHORT_NAMES is not a good name when the class is Constants or Consts. Any improvement is appreciated.

Reply:
Since none of the existing arrays could be reused here because of the different values, the renaming should be in a separate PR, not in this one. In the [JVM specification](https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html#jvms-4.3.2) these short names are introduced as field descriptors, but that name may cause some confusion with the field and it rather refers to the type of the fields.
As far as I can see, this particular "array" contains the primitive or base types, but naming it is the privilege of the author.

Reply:
@nbauma109 any chance you could look at the proposed solution by @JuditKnoll : https://github.com/apache/commons-bcel/pull/221#discussion_r1288233929

Reply:
@garydgregory is it ok now ?

Reply:
@garydgregory any chance you can have a look at it?

Reply:
Sometime this week...

Reply:
@garydgregory can you please look at this PR?